watchtower/wtclient/client.go

```diff
@@ -0,0 +1,722 @@
+	DefaultWriteTimeout = 15 * time.Second
+
+	// DefaultStatInterval specifies the default interval between logging
+	// metrics about the client's operation.
+	DefaultStatInterval = 30 * time.Second
+)
+
+// Client is the primary interface used by the daemon to control a client's
+// lifecycle and backup revoked states.
+type Client interface {
+	// RegisterChannel persistently initializes any channel-dependent
+	// parameters within the client. This should be called during link
+	// startup to ensure that the client is able to support the link during
+	// operation.
+	RegisterChannel(lnwire.ChannelID) error
+
+	// BackupState initiates a request to back up a particular revoked
+	// state. If the method returns nil, the backup is guaranteed to be
+	// successful unless the client is force quit, or the justice
+	// transaction would create dust outputs when trying to abide by the
+	// negotiated policy.
+	BackupState(*lnwire.ChannelID, *lnwallet.BreachRetribution) error
+
+	// Start initializes the watchtower client, allowing it process requests
+	// to backup revoked channel states.
+	Start() error
+
+	// Stop attempts a graceful shutdown of the watchtower client. In doing
+	// so, it will attempt to flush the pipeline and deliver any queued
+	// states to the tower before exiting.
+	Stop() error
+
+	// ForceQuit will forcibly shutdown the watchtower client. Calling this
+	// may lead to queued states being dropped.
+	ForceQuit()
+}
+
+// Config provides the TowerClient with access to the resources it requires to
+// perform its duty. All nillable fields must be non-nil for the tower to be
+// initialized properly.
+type Config struct {
+	// Signer provides access to the wallet so that the client can sign
+	// justice transactions that spend from a remote party's commitment
+	// transaction.
+	Signer input.Signer
+
+	// NewAddress generates a new on-chain sweep pkscript.
+	NewAddress func() ([]byte, error)
+
+	// SecretKeyRing is used to derive the session keys used to communicate
+	// with the tower. The client only stores the KeyLocators internally so
+	// that we never store private keys on disk.
+	SecretKeyRing keychain.SecretKeyRing
+
+	// NetDialer establishes a brontide connection over an onion or clear
+	// network.
+	NetDial NetDialer
+
+	// DB provides access to the client's stable storage medium.
+	DB DB
+
+	// Net specifies the type of network the client will use to initiate
+	// outbound requests, which can be either onion or clear net.
+	Net tor.Net
+
+	// Policy is the session policy the client will propose when creating
+	// new sessions with the tower. If the policy differs from any active
+	// sessions recorded in the database, those sessions will be ignored and
+	// new sessions will be requested immediately.
+	Policy wtpolicy.Policy
+
+	// PrivateTower is the net address of a private tower. The client will
+	// try to create all sessions with this tower.
+	PrivateTower *lnwire.NetAddress
```

Comment:
Not a blocker as is, but this should be updated to allow it to point to multiple towers. Additionally (for our use case elsewhere), we should be able to dynamically add _new_ towers to an existing running client. Based on my current understanding, we should just be able to add the new tower addr to the existing round robin iterator.

Reply:
agreed, i'm planning to do this in a follow up to prevent the scope of this pr from expanding too much